src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -284,10 +357,36 @@ public Uri GetProxy(Uri uri)
                         // we can return the Proxy uri directly.
                         if (proxyInfo.ProxyBypass == IntPtr.Zero)
                         {
-                            string proxyString = Marshal.PtrToStringUni(proxyInfo.Proxy);
-                            ParseProxyConfig(proxyString, out Uri insecureProxy, out Uri secureProxy);
+                            if (proxyInfo.Proxy != IntPtr.Zero)
+                            {
+                                string proxyStr = Marshal.PtrToStringUni(proxyInfo.Proxy);
+
+                                MultiProxy secureProxy, insecureProxy;
+
+                                if (_cachedProxies.TryGetValue(proxyStr, out CachedProxy cached))
+                                {
+                                    cached.LastAccessTicks = Environment.TickCount;
+                                    secureProxy = cached.SecureProxy;
+                                    insecureProxy = cached.InsecureProxy;
```

Comment:
Hmm, I see, this cache is avoiding all the costs of ParseProxyConfig even for the more common single-proxy case?  I still wonder if it's worth the complexity, especially given your previous measurements that showed the costs of ParseProxyConfig pailed in comparison to the overheads of getting and processing the PAC file.

Reply:
Optimization is really just a happy coincidence here; the main reason for doing this is it keeps both single- and multi- on the same code path.

Reply:
Cache removed. Only downside is that now it allocates an identical `Uri` for each PAC-script proxy resolve.